### ISSUE TITLE
aws_ssm_maintenance_window_target name and description fields not optional

### ISSUE THREAD

**Issue by mrobinso12** [terraform-providers/terraform-provider-aws#8791]
<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version

Terraform v0.11.11
* provider.archive: version = "~> 1.2"
* provider.aws: version = "~> 2.12"
* provider.local: version = "~> 1.2"
* provider.null: version = "~> 2.1"
* provider.random: version = "~> 2.1"
* provider.template: version = "~> 2.1"

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* aws_ssm_maintenance_window_target

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
resource "aws_ssm_maintenance_window" "backup" {
  name     = "${var.name_prefix}-nightly-backup"
  schedule = "${var.backup_schedule}"
  duration = 1
  cutoff   = 0
}

resource "aws_ssm_maintenance_window_target" "backup" {
  window_id     = "${aws_ssm_maintenance_window.backup.id}"
  resource_type = "INSTANCE"
  targets {
    key    = "${var.backup_target_tag_key}"
    values = ["${var.backup_target_tag_values}"]
  }
}

resource "aws_iam_role_policy_attachment" "extra" {
  role       = "${var.backup_task_role_name}"
  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonSSMMaintenanceWindowRole"
}

resource "aws_ssm_maintenance_window_task" "backup" {
  name             = "${var.name_prefix}-backup-task"
  description      = "Backup task for ${var.name_prefix}"
  window_id        = "${aws_ssm_maintenance_window.backup.id}"
  task_type        = "RUN_COMMAND"
  task_arn         = "AWS-RunShellScript"
  priority         = 1
  service_role_arn = "${var.backup_task_role_arn}"
  max_concurrency  = "2"
  max_errors       = "1"

  targets {
    key    = "WindowTargetIds"
    values = ["${aws_ssm_maintenance_window_target.backup.id}"]
  }

  task_parameters {
    name   = "commands"
    values = ["${var.backup_commands}"]
  }
}

resource "aws_ssm_association" "backup" {
  name = "AWS-GatherSoftwareInventory"

  targets {
    key    = "${var.backup_target_tag_key}"
    values = ["${var.backup_target_tag_values}"]
  }
}
```

### Expected Behavior

`aws_ssm_maintenance_window_target` should deploy successfully even without `name` and `description` fields, as they are listed as optional.

### Actual Behavior

Receiving following error message without including `name` and `description` fields:

```
* module.datastores.module.data_mgmt.module.scheduled_backups.aws_ssm_maintenance_window_target.backup: 1 error(s) occurred:                                                                  
                                                                                                                                                                                              
* aws_ssm_maintenance_window_target.backup: InvalidParameter: 2 validation error(s) found.                                                                                                    
- minimum field size of 1, RegisterTargetWithMaintenanceWindowInput.Description.                                                                                                              
- minimum field size of 3, RegisterTargetWithMaintenanceWindowInput.Name.                                                                                                                     
```

When adjusting configuration files to include a `name` and `description`, the resources deploy successfully.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply` on config without a `name` and `description` for `aws_ssm_maintenance_window_target`.  This should return error.
2. Add `name` and `description` fields to `aws_ssm_maintenance_window_target` and then run `terraform apply`.  Resources should deploy successfully.

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

We have previously deployed these resources without issue.  This has just started within the past few days.

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* https://github.com/terraform-providers/terraform-provider-aws/pull/8671

**Comment by Dkochkerian** [comment]
Breaks starting 2.12.0, similar issue with aws_ssm_maintenance_window_task was resolved a few months ago https://github.com/terraform-providers/terraform-provider-aws/issues/6715

**Comment by fabriciomariani** [comment]
Same is happening here. Running "aws" (2.16.0).

```
Error: Error applying plan:

1 error(s) occurred:

* module.automaintenance-window-dev-1.aws_ssm_maintenance_window_target.target: 1 error(s) occurred:

* aws_ssm_maintenance_window_target.target: InvalidParameter: 2 validation error(s) found.
- minimum field size of 1, RegisterTargetWithMaintenanceWindowInput.Description.
- minimum field size of 3, RegisterTargetWithMaintenanceWindowInput.Name.
```

**Comment by stack72** [comment]
PR incoming! It was an easy fix

**Comment by bflad** [comment]
The fix for this issue on resource creation has been merged and will release with version 2.17.0 of the Terraform AWS Provider, likely later today. Thanks to @stack72 for the remediation.

**Comment by bflad** [comment]
This has been released in [version 2.17.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2170-june-28-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!
